x-pack/heartbeat/monitors/browser/synthexec/synthexec.go

```diff
@@ -198,7 +205,9 @@ func runCmd(
 	// Send the test results into the output
 	wg.Add(1)
 	go func() {
-		defer jsonReader.Close()
+		defer func() {
+			_ = jsonReader.Close()
+		}()
```

Comment:
Can't we defer as we were doing previously?

I tried to do some research to understand the difference between this and the previous `defer` pattern and I couldn't find anything that would explain it.

[I found this response on SO interesting](https://stackoverflow.com/questions/32541870/multiple-defers-vs-deferred-anonymous-function) but it actually advocates for the opposite pattern: directly deferring (although I can see the advantage of a lambda if we're doing any error handling for the deferred function.

[Maybe you're doing it to evaluate `jsonReader`](https://yourbasic.org/golang/defer/) only when the function exits? I couldn't see why that would be helpful in this case though since we only assign to it with `os.Pipe()` at the beginning of the function?

If this is indeed intentional perhaps it would be worth leaving a comment as to why we do it.

Reply:
Same story WRT the linter and errors